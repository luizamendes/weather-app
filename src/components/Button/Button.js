import React from "react";
import { Link } from "react-router-dom";
import "./index.css";

export const Button = ({ children, link, ...props }) => {
  if (link) {
    return (
      <Link {...props} className="button" to={link}>
        {children}
      </Link>
    );
  }
  return (
    <button {...props} className="button">
      {children}
    </button>
  );
};
