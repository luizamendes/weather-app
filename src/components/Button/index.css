.button {
  height: 3rem;
  margin: 1rem auto 1rem;
  padding: 0.5rem 3rem;
  background-color: #ff0000;
  border: 0;
  border-radius: 0.5rem;
  color: white;
  font-weight: bold;
  font-size: 1.2em;
  cursor: pointer;
}

.button:hover {
  background-color: #ff8c00;
}

.button:disabled,
.button:disabled:hover {
  background-color: #999;
  cursor: not-allowed;
}

a.button {
  display: block;
  width: auto;
  height: auto;
  font-size: 1em;
  text-decoration: none;
  text-align: center;
}
