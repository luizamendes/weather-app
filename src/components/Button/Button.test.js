import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Button } from './Button';

describe('Buttom component Snapshot', () => {
  it('renders a button', () => {
    const tree = renderer
      .create(
        <Router>
          <Button>Click Me</Button>
        </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders a anchor tag', () => {
    const tree = renderer
      .create(
        <Router>
          <Button link="/something">Click Me</Button>
        </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('Testing functionality', () => {
  it('should execute onClick function when clicked', () => {
    const handleOnClick = jest.fn();

    render(
      <Router>
        <Button onClick={handleOnClick}>Click me</Button>
      </Router>
    );

    fireEvent.click(screen.getByText('Click me'));
    expect(handleOnClick).toHaveBeenCalledTimes(1);
  });
});
