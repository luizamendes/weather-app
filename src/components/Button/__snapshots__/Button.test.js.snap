// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttom component Snapshot renders a anchor tag 1`] = `
<a
  className="button"
  href="/something"
  onClick={[Function]}
>
  Click Me
</a>
`;

exports[`Buttom component Snapshot renders a button 1`] = `
<button
  className="button"
>
  Click Me
</button>
`;
